### PR TITLE
fix(install/helm): envoy gateway configuration propagation from values file to config map

### LINKED ISSUE #1607: Pass envoy gateway config from values.yaml to helm chart configmap
*Description*:
>What issue is being seen? Describe what should be happening instead of
the bug, for example: Envoy should not crash, the expected value isn't
returned, etc.

The values configured in values.yaml for configuring envoy gateway config are not propagated to `envoy-gateway-config.yaml`.

*Repro steps*:
> Include sample requests, environment, etc. All data and inputs
required to reproduce the bug.

Install helm chart using values.yaml with envoyGateway configuration as below:
```yaml
config:
  envoyGateway:
    gateway:
      controllerName: gateway.envoyproxy.io/gatewayclass-controller
    provider:
      type: Kubernetes
      kubernetes:
        rateLimitDeployment:
          replicas: 2
    rateLimit:
      backend:
        type: Redis
        redis:
          url: redis.redis-system.svc.cluster.local:6379
```

The expected behavior is the propagation of the above configuration to `envoy-gateway-config.yaml` config map but only the controllerName and provider.type are propagated to the install config map manifest.

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Environment*:
>Include the environment like gateway version, envoy version and so on.

*Logs*:
>Include the access logs and the Envoy logs.

### PR DESCRIPTION
**Pull request by psinghal20**
**What this PR does / why we need it**:
This PR updates the config map manifest in the helm chart to propagate the envoyGateway config map from values file.

**Which issue(s) this PR fixes**:

Fixes #1607

### CODE REVIEW

File: charts/gateway-helm/templates/envoy-gateway-config.yaml [hunk 1]
```diff
@@ -13,7 +13,15 @@ data:
         | quote }}
     kind: EnvoyGateway
     provider:
-      type: {{ .Values.config.envoyGateway.provider.type | quote }}
+      {{- toYaml .Values.config.envoyGateway.provider | nindent 6 }}
```
Comment (zirain) [inline charts/gateway-helm/templates/envoy-gateway-config.yaml:16]: My concern is everytime there's an new field introduced, you need update these lines too.
Is it possible render `Values.config.envoyGateway` directly?

Comment (psinghal20) [reply charts/gateway-helm/templates/envoy-gateway-config.yaml]: Hi @zirain, I have updated the manifest file to render the complete `values.config.envoyGateway` configuration and moved the logging levels to the values file as well.

Comment (zirain) [reply charts/gateway-helm/templates/envoy-gateway-config.yaml]: would you like to add an e2e test to ensure this work as expected?

Comment (psinghal20) [reply charts/gateway-helm/templates/envoy-gateway-config.yaml]: Sure, I would love to. 

WDYT about updating the current ratelimit tests manifests [here](https://github.com/envoyproxy/gateway/blob/main/examples/redis/redis.yaml#L52-L69) with a Kubernetes provider deployment spec(something like configuring cache key prefix) and checking the desired behavior in ratelimit e2e tests?

Comment (zirain) [reply charts/gateway-helm/templates/envoy-gateway-config.yaml]: you can reuse that manifests configuration, but in a sepreated test case.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]
LGTM, a e2e in following PR will be better.

**Pull request merged**